provider/lib/app.dart

```diff
@@ -0,0 +1,49 @@
+// Copyright 2018 The Flutter Architecture Sample Authors. All rights reserved.
+// Use of this source code is governed by the MIT license that can be found
+// in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:meta/meta.dart';
+import 'package:provider/provider.dart';
+import 'package:provider_sample/add_todo_screen.dart';
+import 'package:provider_sample/localization.dart';
+import 'package:provider_sample/todo_list_model.dart';
+import 'package:todos_app_core/todos_app_core.dart';
+import 'package:todos_repository_core/todos_repository_core.dart';
+
+import 'home/home_screen.dart';
+
+class ProviderApp extends StatelessWidget {
+  final TodosRepository repository;
+
+  ProviderApp({
+    @required this.repository,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return ChangeNotifierProvider(
```

Comment:
As a follow up to the comment on `main.dart` this probably should be updated.

So either:

```dart
ChangeNotifierProvider(
  create: (context) => TodoListModel(repository: Provider.of<TodosRepository>(context, listen: false),
)
```

or:

```dart
ChangeNotifierProxyProvider<TodosRepository, TodoListModel>(
  create: (context) => TodoListModel(),
  update: (_, todosRepository, model) => model..todosRepository = todosRepository,
)
```

Your call